File: app/src/main/java/com/jds/fitnessjunkiess/getfitapp/Data/ViewModels/WorkoutsViewModel.java
package com.jds.fitnessjunkiess.getfitapp.Data.ViewModels;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;

import com.jds.fitnessjunkiess.getfitapp.Data.Entities.Workout;
import com.jds.fitnessjunkiess.getfitapp.Data.Repositories.WorkoutsRepository;

import java.util.List;

public class WorkoutsViewModel extends AndroidViewModel {
  WorkoutsRepository repository;
  LiveData<List<Workout>> data;

  public WorkoutsViewModel(Application application) {
    super(application);
    this.repository = new WorkoutsRepository(application);
    this.data= this.repository.getData();
  }

  public LiveData<List<Workout>> getData() {
    return data;
  }

  public void insert(Workout workout) {
    this.repository.insert(workout);
  }
}

File: app/src/main/java/com/jds/fitnessjunkiess/getfitapp/data/viewModels/WorkoutsViewModel.kt
package com.jds.fitnessjunkiess.getfitapp.data.viewModels

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.LiveData

import com.jds.fitnessjunkiess.getfitapp.data.dataModels.Workout
import com.jds.fitnessjunkiess.getfitapp.data.repositories.WorkoutsRepository

class WorkoutsViewModel(application: Application) : AndroidViewModel(application) {
    internal var repository: WorkoutsRepository = WorkoutsRepository(application)
    var data: LiveData<List<Workout>>
        internal set

    init {
        this.data = this.repository.data
    }

    fun insert(workout: Workout) {
        this.repository.insert(workout)
    }
}
